splash/browser_tab.py

```diff
@@ -1078,6 +1093,65 @@ def log(self, message, min_level=None):
         log.msg(message, system='render')
 
 
+class ElementsStorage(QObject):
+    def __init__(self, parent):
+        self.name = str(uuid.uuid1())
+        super(ElementsStorage, self).__init__(parent)
+
+    @pyqtSlot(name="getId", result=str)
+    def get_id(self):
+        return str(uuid.uuid1())
+
+
+class HTMLElement(object):
+    DIMENSIONS_JS_FUNC = """
+        (function(element) {
+            var rect = element.getBoundingClientRect();
```

Comment:
Yes, exactly, both of the methods exist for a reason. It might also be
worth it to perform a hittest on the coordinates and raise an error if its
not successful. The element might be hidden behind another element or
transparent in that coordinates, so raising an error would be better than
failing silently and possibly causing side effects.
On Jul 7, 2016 12:08, "dennis again" <notifications@github.com> wrote:

> In splash/browser_tab.py
> <https://github.com/scrapinghub/splash/pull/471#discussion_r69851779>:
>
> > @@ -1078,6 +1093,65 @@ def log(self, message, min_level=None):
> >          log.msg(message, system='render')
> >
> >
> > +class ElementsStorage(QObject):
> > +    def __init__(self, parent):
> > +        self.name = str(uuid.uuid1())
> > +        super(ElementsStorage, self).__init__(parent)
> > +
> > +    @pyqtSlot(name="getId", result=str)
> > +    def get_id(self):
> > +        return str(uuid.uuid1())
> > +
> > +
> > +class HTMLElement(object):
> > +    DIMENSIONS_JS_FUNC = """
> > +        (function(element) {
> > +            var rect = element.getBoundingClientRect();
>
> That might be a problem if you want to "click" the element. If you want to
> take a screenshot, you most likely won't be satisfied with just the first
> client rect.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/scrapinghub/splash/pull/471/files/37b17cdce31ab27b2b866f6afa13b69f2607c059#r69851779>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AANrvUUUhqQV2uQ65VdjchSgpWpEm0oTks5qTInngaJpZM4I8Dce>
> .
>